### PR TITLE
[PackageLoading] Improve error message when a custom target path does not exist

### PR DESCRIPTION
**Pull request by ssuuyyuu**
The issue is that even when a custom target path is specified in manifest,
it still shows an error message asking to specify a path. The fix is to check
if a path is already specified but does not exist, then it indicates the path
dose not exist. 

A test is added for the fix.

Resolves [SR-9558](https://bugs.swift.org/browse/SR-9558).

### CODE REVIEW

**Comment by ssuuyyuu** [comment]
@aciidb0mb3r Hi Ankit, would you help me review this PR, please? Thank you!

File: Sources/PackageLoading/PackageBuilder.swift [hunk 1]
```diff
@@ -27,6 +27,9 @@ public enum ModuleError: Swift.Error {
     /// One or more referenced targets could not be found.
     case modulesNotFound([String])
 
+    /// Invalid custom path
+    case invalidCustomPath((target: String, path: String))
```
Comment (hartbit) [inline Sources/PackageLoading/PackageBuilder.swift]: Two enum associated values feel more natural than an anonymous tuple
```suggestion
    case invalidCustomPath(target: String, path: String)
```

**Comment by hartbit** [comment]
@swift-ci please smoke test

**Comment by hartbit** [comment]
@aciidb0mb3r Are you okay with merging this?

File: Sources/PackageLoading/PackageBuilder.swift [hunk 2]
```diff
@@ -27,6 +27,9 @@ public enum ModuleError: Swift.Error {
     /// One or more referenced targets could not be found.
     case modulesNotFound([String])
 
+    /// Invalid custom path
```
Comment (aciidb0mb3r) [inline Sources/PackageLoading/PackageBuilder.swift]: nit: end comment with a full stop (.)

File: Sources/PackageLoading/PackageBuilder.swift [hunk 3]
```diff
@@ -91,6 +94,8 @@ extension ModuleError: CustomStringConvertible {
             return "target '\(target)' in package '\(package)' is outside the package root"
         case .unsupportedTargetPath(let targetPath):
             return "target path '\(targetPath)' is not supported; it should be relative to package root"
+        case .invalidCustomPath(let target, let path):
+            return "target path '\(path)' for target '\(target)' does not exist"
```
Comment (aciidb0mb3r) [inline Sources/PackageLoading/PackageBuilder.swift]: maybe reword to `"invalid custom path '\(path)' for target '\(target)'"`

File: Tests/PackageLoadingTests/PackageBuilderTests.swift [hunk 4]
```diff
@@ -1100,6 +1100,23 @@ class PackageBuilderTests: XCTestCase {
...
+            name: "Foo",
+            targets: [
+                TargetDescription(name: "Foo", path: "./NotExist")
+                ]
+        )
+
+        PackageBuilderTester(manifest, in: fs) { result in
+
+            result.checkDiagnostic("target path './NotExist' for target 'Foo' does not exist")
+
```
Comment (aciidb0mb3r) [inline Tests/PackageLoadingTests/PackageBuilderTests.swift]: nit: unnecessary new line

File: Tests/PackageLoadingTests/PackageBuilderTests.swift [hunk 5]
```diff
@@ -1100,6 +1100,23 @@ class PackageBuilderTests: XCTestCase {
...
+        let manifest = Manifest.createV4Manifest(
+            name: "Foo",
+            targets: [
+                TargetDescription(name: "Foo", path: "./NotExist")
+                ]
+        )
+
+        PackageBuilderTester(manifest, in: fs) { result in
+
```
Comment (aciidb0mb3r) [inline Tests/PackageLoadingTests/PackageBuilderTests.swift]: nit: unnecessary new line

File: Tests/PackageLoadingTests/PackageBuilderTests.swift [hunk 6]
```diff
@@ -1100,6 +1100,23 @@ class PackageBuilderTests: XCTestCase {
...
     }
 
+    func testSpecifiedCustomPathDoesNotExist() {
+        let fs = InMemoryFileSystem(emptyFiles: "/Foo.swift")
+
+        let manifest = Manifest.createV4Manifest(
+            name: "Foo",
+            targets: [
+                TargetDescription(name: "Foo", path: "./NotExist")
+                ]
```
Comment (aciidb0mb3r) [inline Tests/PackageLoadingTests/PackageBuilderTests.swift]:
```suggestion
            ]
```

**Comment by ssuuyyuu** [comment]
@aciidb0mb3r Thanks for the comments! I have updated accordingly.
